pkg/controller/xgboostjob/expectation.go

```diff
@@ -62,7 +63,14 @@ func onDependentCreateFunc(r reconcile.Reconciler) func(event.CreateEvent) bool
 
 		logrus.Info("Update on create function ", xgbr.ControllerName(), " create object ", e.Meta.GetName())
 		if controllerRef := metav1.GetControllerOf(e.Meta); controllerRef != nil {
-			expectKey := job_controller.GenExpectationPodsKey(e.Meta.GetNamespace()+"/"+controllerRef.Name, rtype)
+			var expectKey string
+			if _, ok := e.Object.(*corev1.Pod); ok {
+				expectKey = job_controller.GenExpectationPodsKey(e.Meta.GetNamespace()+"/"+controllerRef.Name, rtype)
```

Comment:
Move `e.Meta.GetNamespace()+"/"+controllerRef.Name` to a variable that can be reused?

Reply:
Yeah, sounds good. I will update it